tools/onenote-migration/onenote_migrate.py

```
@@ -0,0 +1,621 @@
+    app = _MSAL(cfg.CLIENT_ID, authority=cfg.AUTHORITY, token_cache=cache)
+
+    # Try silent acquisition first (uses cached refresh token)
+    accounts = app.get_accounts()
+    result = None
+    if accounts:
+        result = app.acquire_token_silent(cfg.SCOPES, account=accounts[0])
+
+    if not result:
+        flow = app.initiate_device_flow(scopes=cfg.SCOPES)
+        if "user_code" not in flow:
+            raise RuntimeError(f"Device flow failed: {flow.get('error_description', flow)}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)